code/modules/projectiles/guns/projectile/rocketlauncher.dm

```diff
@@ -53,6 +53,64 @@
 			return(SUICIDE_ACT_BRUTELOSS)
 	return
 
+/obj/item/weapon/gun/projectile/rocketlauncher/nanotrasen
+	name = "rocket launcher"
+	desc = "Watch the backblast, you idiot."
+	fire_sound = 'sound/weapons/rocket.ogg'
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns.dmi', "right_hand" = 'icons/mob/in-hand/right/guns.dmi')
+	icon_state = "rpg_nt"
+	item_state = "rpg_nt"
+	max_shells = 1
+	w_class = W_CLASS_LARGE
+	starting_materials = list(MAT_IRON = 5000)
+	w_type = RECYK_METAL
+	force = 10
+	recoil = 5
+	throw_speed = 4
+	throw_range = 3
+	fire_delay = 5
+	flags = FPRINT
+	siemens_coefficient = 1
+	slot_flags = SLOT_BACK
+	caliber = list(ROCKETGRENADE = 1)
+	origin_tech = Tc_COMBAT + "=4;" + Tc_MATERIALS + "=2;"
+	ammo_type = "/obj/item/ammo_casing/rocket_rpg/foam"
+	attack_verb = list("strikes", "hits", "bashes")
+	gun_flags = 0
+
+/obj/item/weapon/gun/projectile/rocketlauncher/nanotrasen/isHandgun()
+	return FALSE
+
+/obj/item/weapon/gun/projectile/rocketlauncher/nanotrasen/update_icon()
+	if(!getAmmo())
+		icon_state = "rpg_nt_e"
+		item_state = "rpg_nt_e"
+	else
+		icon_state = "rpg_nt"
+		item_state = "rpg_nt"
+
+/obj/item/weapon/gun/projectile/rocketlauncher/nanotrasen/attack(mob/living/M as mob, mob/living/user as mob, def_zone)
+	if(M == user && user.zone_sel.selecting == "mouth") //Are we trying to suicide by shooting our head off ?
+		user.visible_message("<span class='warning'>[user] tries to fit \the [src] into \his mouth but quickly reconsiders it</span>", \
+		"<span class='warning'>You try to fit \the [src] into your mouth. You feel silly and pull it out</span>")
+		return // Nope
+	..()
+
+/obj/item/weapon/gun/projectile/rocketlauncher/nanotrasen/suicide_act(var/mob/user)
+	if(!src.process_chambered()) //No rocket in the rocket launcher
+		user.visible_message("<span class='danger'>[user] jams down \the [src]'s trigger before noticing it isn't loaded and starts bashing \his head in with it! It looks like \he's trying to commit suicide.</span>")
+		return(SUICIDE_ACT_BRUTELOSS)
+	else //Needed to get that shitty default suicide_act out of the way
+		user.visible_message("<span class='danger'>[user] fiddles with \the [src]'s safeties and suddenly aims it at \his feet! It looks like \he's trying to commit suicide.</span>")
+		spawn(10) //RUN YOU IDIOT, RUN
+			explosion(src.loc, -1, 1, 4, 8)
+			if(src) //Is the rocket launcher somehow still here ?
+				qdel(src) //This never happened
+			return(SUICIDE_ACT_BRUTELOSS)
+	return
```

Comment:
If this and the `attack()` are the same as the normal RPG, you don't have to define them